day4_2018.py

```diff
@@ -0,0 +1,94 @@
+with open("input4.txt", "r") as f:
+    inputs = f.read().split("\n")
+    inputs.sort()
+    inputs = [x.split(" ") for x in inputs]
+
+
+def get_guards_ids(inputs):
+    guards_id = []
+    for data_line in inputs:
+        if len(data_line) == 6 and data_line[3][1:] not in guards_id:
+            guards_id.append(data_line[3][1:])
+    return guards_id
+
+
+def get_length_of_next_asleep(idx):
+    guard_asleep_minutes = 0
+    specific_asleep_minutes = []
+    if idx < len(inputs) - 3:
+        wake_up = inputs[idx + 4][1]
+        asslep = inputs[idx + 3][1]
+        if inputs[idx + 3][3] == "asleep":
+            length_of_sleep = int(wake_up[3:5]) - int(asslep[3:5])
+            guard_asleep_minutes += length_of_sleep
+            for minutes in range(int(asslep[3:5]), int(wake_up[3:5])):
+                specific_asleep_minutes.append(minutes)
```

Comment:
```suggestion
        wake_up = int(inputs[idx + 4][1][3:5])
        asleep = int(inputs[idx + 3][1][3:5])
        if inputs[idx + 3][3] == "asleep":
            specific_asleep_minutes = list(range(asleep, wake_up))
            guard_asleep_minutes = len(specific_asleep_minutes)
```
Niepotrzebne są te powtórzenia z indeksowaniem [3:5], można to uprościć od razu przypisując do zmiennej + jakieś tam inne uproszczenia.